prescient.el

```diff
@@ -119,6 +119,25 @@ be `literal+initialism', which equivalent to the list (`literal'
           (const :tag "Prefix" prefix)
           (const :tag "Anchored" anchored)))
 
+(defcustom prescient-filter-alist
+  '((literal . prescient--literal-regexp)
+    (initialism . prescient--initials-regexp)
+    (regexp . prescient--regexp-regexp)
+    (fuzzy . prescient--fuzzy-regexp)
+    (prefix . prescient--prefix-regexp)
+    (anchored . prescient--anchored-regexp))
+  "An alist of filter methods and their functions.
+
+These symbols can be included `prescient-filter-method', and
+their corresponding functions will be used to create regexps for
+matching candidates.
+
+A function should take two arguments: the query for which it is
+to create a regexp, and a boolean that describes whether it
+should enclose matched text in capture groups (such as with
+`prescient--with-group')."
```

Comment:
I think it would be good to also mention the difference in behavior if the boolean is the symbol `all` (see the docstring of `prescient-filter-regexps`).